python/cudf/cudf/bindings/cudf_cpp.pyx

```diff
@@ -163,8 +268,16 @@ cdef set_scalar_value(gdf_scalar *scalar, val):
         scalar.data.si08 = val
     elif val.dtype.type == np.bool or val.dtype.type == np.bool_:
         scalar.data.b08 = val
-    elif val.dtype.name == 'datetime64[ms]':
-        scalar.data.dt64 = np.datetime64(val, 'ms').astype(int)
+    elif val.dtype.type == np.datetime64:
+        time_unit, _ = np.datetime_data(val.dtype)
+        if time_unit == 'ms':
+            scalar.data.dt64 = np.datetime64(val, time_unit).astype(int)
+        elif time_unit in np_to_gdf_time_unit:
+            scalar.data.tmst = np.datetime64(val, time_unit).astype(int)
+        elif time_unit == 'D':
+            scalar.data.dt32 = np.datetime64(val, time_unit).astype(int)
+        else:
+            scalar.data.dt64 = np.datetime64(val, time_unit).astype(int)
```

Comment:
You're typecasting from 64bits to 32bits here. I wonder if you can just pass val and it will handle it as an int64 scalar or if you'd need to view it as an int64 first, but regardless you should be able to just do `np.int64(val)` here.

Reply:
Updated in https://github.com/rapidsai/cudf/pull/2213/commits/50f6e50cd507e70530631f536ddd3d1c2eca731a.